src/plugins/vis_type_timeseries/public/application/components/index_pattern.js

```diff
@@ -119,10 +120,9 @@ export const IndexPattern = ({ fields, prefix, onChange, disabled, model: _model
 
   const model = { ...defaults, ..._model };
 
-  const dataSourceManagementEnabled =
-    !!getDataSourceManagementSetup().dataSourceManagement || false;
+  const mdsEnabled = !!getDataSourceSetup().dataSource || false;
   const handleDataSourceSelectChange = createDataSourcePickerHandler(onChange);
-  const DataSourceSelector = dataSourceManagementEnabled
+  const DataSourceSelector = mdsEnabled
```

Comment:
any reason we change the checking logic here?

Reply:
We have changed the behavior of the datasource management plugin's registration: we will always register the dsm plugin, and the feature flag will be only reflected on these conditional renders for MDS component. According to this, instead of checking if the mds plugin is presented, we should check the data source plugin. (cc: @BionIT )